components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2531,15 +2566,34 @@ private void overrideResidentIdpEPUrls(IdentityProvider residentIDP)
             throws IdentityProviderManagementException {
 
         // Not all endpoints are persisted. So we need to update only a few properties.
-
-        String samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true,
-                true, true, false);
+        String samlSSOUrl;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                samlSSOUrl = ServiceURLBuilder.create().addPath(IdentityConstants.ServerConfig.SAMLSSO).build()
+                        .getAbsoluteURL();
+            } catch (URLBuilderException e) {
+                throw IdentityRuntimeException
+                        .error("Error while building URL: " + IdentityConstants.ServerConfig.SAMLSSO, e);
+            }
+        } else {
+            samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true);
+        }
```

Comment:
Same is done above right ?

Reply:
this is a done for a different method.

Reply:
ok
So all these paths just send the path and construct from old way looks to me we can have a method that will just take the context and URL for the legacy mode and return
So handling the exception does not needs to be duplicated, checking tenant qualified does not needs to be duplicated and readability can be improved

Reply:
Introduced a method to handle this.

Reply:
This logic should not ideally be here. Rather if this is something that should not be persisted it should be fixed at that place rather than overriding again after reading properties from DB.
Yet, in the context of present scope fixing this path is fine.